add dsn, domain and evm support

DaMandal0rian opened the pull request:
This PR add support for multiple domains such as core-evm and introduces DSN node key support.
- change provider for infrastructure
- add capability to launch a domain i.e evm-core or payments
- add dsn bootstrap node

DaMandal0rian committed:
add dsn, domain and evm support

DaMandal0rian committed:
update gitignore

vedhavyas reviewed (commented):
The changes seem to introduce EVM along side Payments. This does not work since node does not support running both domains. The ideal final state we want is to update the current scripts to start a single domain but introduce more confiugurations. Example: we have `payments` built into scripts since there was not other domain available when then. Now we have multiple domains. We need to make those configurable like we did DomainID

DaMandal0rian committed:
make domains configurable

DaMandal0rian committed:
minor adjustments with output and variables

DaMandal0rian committed:
configuration changes

- rename hostname for instances to match dns
- change node images used for node key generation
- change IP address output variable names

DaMandal0rian committed:
Merge branch 'main' into gemini-network-domains

DaMandal0rian commented on network-primitives/variables.tf:
```diff
@@ -50,7 +50,26 @@ variable "rpc-node-config" {
+}
+
+variable "domain-node-config" {
+  description = "RPC node deployment config"
```
typo to fix

DaMandal0rian commented on resources/gemini-3d/main.tf:
```diff
@@ -16,24 +16,41 @@ module "gemini-3d" {
+    reserved-only       = false
+    prune               = false
+    node-dsn-port       = 30434
+    enable-domains      = false
```
does `enable-domains` need to be configurable on rpc node too?

vedhavyas replied:
We need to remove all the domain related stuff from RPC including System domain. Seems like we still have system domain running on RPC

DaMandal0rian commented on network-primitives/scripts/create_domain_node_compose_file.sh:
```diff
@@ -0,0 +1,168 @@
+        echo '      "--base-path", "/var/subspace/core_evm_domain",'
+        echo '      "--keystore-path", "/var/subspace/keystore",'
+        echo '      "--rpc-cors", "all",'
+        echo '      "--rpc-port", "6933",'
```
fix port to match caddy

DaMandal0rian committed:
Minor fixes

- change port of domain to match caddy
- fix typo
- fix hostnames

vedhavyas commented on network-primitives/scripts/create_domain_node_compose_file.sh:
```diff
@@ -0,0 +1,168 @@
+      - "30333:30333"
+      - "\${NODE_DSN_PORT}:30433"
+    labels:
+      caddy_0: \${DOMAIN_PREFIX}-\${NODE_ID}.\${NETWORK_NAME}.subspace.network
```
i dont think we need to expose the primary rpc here. It is already exposed under rpc node anyway

vedhavyas commented on network-primitives/scripts/create_domain_node_compose_file.sh:
```diff
@@ -0,0 +1,168 @@
+      echo '      "--no-private-ipv4",'
+      echo '      "--unsafe-ws-external",'
+      echo "      \"--relayer-id=\$RELAYER_SYSTEM_ID\","
+      if [ "${domain_id}" == 1 ]; then
```
no, this is not the way. 
We made it configurable so we can avoid these if else. We can totally remove these IMO

vedhavyas reviewed (commented):
Looks good. There are still domain related infra under RPC. We should remove those as well

DaMandal0rian committed:
domain adjustments

- remove system domain from rpc completely
- fix domain docker compose add system domain
- rpc docker configuration changes
- fix variables
- fix caddy ports
- reset devnet
- hostname fixes

vedhavyas commented on network-primitives/bootstrap_node_provisioner.tf:
```diff
@@ -156,7 +174,7 @@ resource "null_resource" "start-boostrap-nodes" {
       "systemctl reenable subspace.service",
 
       # set hostname
-      "hostnamectl set-hostname ${var.network-name}-bootstrap-node-${count.index}",
```
why change the hostname for bootstrap node with domain prefix ?

DaMandal0rian replied:
I am trying to make the hostnames as close as possible to dns and also machine names in provider dashboard.

vedhavyas replied:
Sure but this not represent any closer TBH. Domain prefix is used for RPC 
Are you trying to set the where node is located on hetzner?

vedhavyas commented on network-primitives/dns.tf:
```diff
@@ -18,10 +18,10 @@ resource "cloudflare_record" "system-domain-rpc" {
 
-resource "cloudflare_record" "core-payments-domain-rpc" {
+resource "cloudflare_record" "core-domain-rpc" {
   count   = length(local.rpc_node_ip_v4)
```
this is still referring to rpc instead of domain ip_v4

vedhavyas commented on network-primitives/farmer_node_provisioner.tf:
```diff
@@ -156,7 +156,7 @@ resource "null_resource" "start-farmer-nodes" {
       "systemctl reenable subspace.service",
 
       # set hostname
-      "hostnamectl set-hostname ${var.network-name}-farmer-node-${count.index}",
```
not sure this change is required here

vedhavyas commented on network-primitives/full_node_provisioner.tf:
```diff
@@ -156,7 +156,7 @@ resource "null_resource" "start-full-nodes" {
       "systemctl reenable subspace.service",
 
       # set hostname
-      "hostnamectl set-hostname ${var.network-name}-full-node-${count.index}",
```
Same as above. Why change the hostnames of primary with domain specific details ?

vedhavyas commented on network-primitives/rpc_node_provisioner.tf:
```diff
@@ -168,7 +168,7 @@ resource "null_resource" "start-rpc-nodes" {
       "systemctl reenable subspace.service",
 
       # set hostname
-      "hostnamectl set-hostname ${var.network-name}-rpc-node-${count.index}",
```
Same as above

DaMandal0rian replied:
I am trying to make the hostnames as close as possible to dns and also machine names in provider dashboard.

vedhavyas reviewed (commented):
Some minor nits. 
We are good to merge after. Thanks!
